src/System.Tuples/src/System/Tuples/ValueTuple.cs

```diff
@@ -0,0 +1,1205 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace System
+{
+    /// <summary>
+    /// Helper so we can call some tuple methods recursively without knowing the underlying types.
+    /// </summary>
+    internal interface ITupleInternal
+    {
+        string ToString(StringBuilder sb);
```

Comment:
Feels like this should just be ```void AppendTo(StringBuilder sb)``` and the caller should call ToString() on the result.